src/test/java/StringManipulationTest.java

```diff
@@ -50,100 +59,152 @@ public void testRemoveNthCharacter1() {
+                manipulatedstring.removeNthCharacter(5, false));
     }
 
+    // we test for a smaller n.
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("let's try a smaller string");
+        assertEquals("ltstyasalrsrn",
+                manipulatedstring.removeNthCharacter(2, false));
     }
 
+    // we test for a smaller n while maintaning the spacing in the string
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("let's try a smaller string");
+        assertEquals("l t s t y a s a l r s r n ",
+                manipulatedstring.removeNthCharacter(2, true));
     }
 
+    // test to make sure it throws a IndexOutOfBoundsException when n > string
+    // length
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This Test should throw IndexOutOfBoundsException for n > string length");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.removeNthCharacter(100, false));
     }
 
+    // test to make sure it throws a IllegalArgumentException when n < 0
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This Test should throw IllegalArgumentException for n < 0");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.removeNthCharacter(-1, false));
     }
 
     @Test
     public void testGeSubStrings1() {
         manipulatedstring.setString("This is my string");
-        String [] sStings = manipulatedstring.getSubStrings(3, 4);
+        String[] sStings = manipulatedstring.getSubStrings(3, 4);
 
         assertEquals(sStings[0], "my");
         assertEquals(sStings[1], "string");
     }
 
+    // test to make sure that it can scale to more substrings
     @Test
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is a very long string that seems to go on and on and on");
+        String[] sStings = manipulatedstring.getSubStrings(2, 7);
+
+        assertEquals(sStings[0], "is");
+        assertEquals(sStings[1], "a");
+        assertEquals(sStings[2], "very");
+        assertEquals(sStings[3], "long");
+        assertEquals(sStings[4], "string");
+        assertEquals(sStings[5], "that");
     }
+
+    // tests thats startWord cannot be <= 0 and should throw a
+    // IllegalArgumentException
     @Test
     public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This Test should throw IllegalArgumentException for startWord <= 0");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.getSubStrings(0, 5));
     }
+
+    // tests thats endWord cannot be <= 0 and should throw a
+    // IllegalArgumentException
```

Comment:
Modify this test case so that startWord is an integer greater than 0. This ensures that the exception is caused by the endWord <= 0, not because startWord is <= 0.